src/bucket/BucketListSnapshot.cpp

```diff
@@ -0,0 +1,292 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
```

Comment:
Looks like there are some outdated files in the beginning of the commit history, and some small fixes at the end, please squash the commits (commit history doesn't need to be super precise, just make sure to remove obsolete interface changes etc)